src/main.rs

```diff
@@ -59,31 +80,38 @@ fn run() -> Result<(), Box<dyn Error>> {
     }
 }
 
+fn wait_for_child(mut child: Child) -> Result<(), Box<dyn Error>> {
+    let status = child.wait()?;
+    if status.success() {
+        Ok(())
+    } else {
+        Err(Box::new(FailureStatus(status)))
```

Comment:
We should close the process with the same status code with which the subprocess exited.